### ISSUE TITLE
Issue about "model.train()" and "model.eval()"??

### ISSUE THREAD

**Issue by hpkhanh1610** [udacity/deep-learning-v2-pytorch#3]
I have a question about model.train() and model.eval().

As far as I see in lecture content video by Cezanne together with the notebooks, for example, "_mnist_mlp_solution_with_validation_" or "_cifar10_cnn_augmentation_", why don't we use **model.eval()** for the validation and test time and **model.train()** for training? We have dropout in the model, so I think we should switch between 2 modes.

In the lecture video of _"Deep learning with pytorch"_ by Mat Leonard, in notebook "_Part 5 - Inference and Validation_" in this link _https://github.com/udacity/DL_PyTorch_, I see that Mat switches **model.train()** and **model.eval()** during training and validating. I asked my mentor and he asked me to raise an issue here.

Thank you.

**Comment by cezannec** [comment]
Hello! You're absolutely right. Any model with dropout or batch normalization will have different behavior during training and evaluation. This was an oversight of mine and I've just updated the code to include the two cases. Thank you for catching this!
